backend-api/src/functions/asyncActiveSession/tokenService/tokenService.ts

```diff
@@ -52,6 +51,38 @@ export class TokenService implements ITokenService {
     return successResult("");
   };
 
+  private async fetchPublicKeyWithRetries(
+    stsJwksEndpoint: string,
+  ): Promise<Result<JSON>> {
+    const maxRetries = 2;
+    const delayInMs = 1000;
+
+    let fetchPublicKeyResult: Result<JSON> | undefined;
+
+    for (let retries = 0; retries <= maxRetries; retries++) {
+      fetchPublicKeyResult = await this.fetchPublicKey(stsJwksEndpoint);
+
+      if (!fetchPublicKeyResult.isError) {
+        return fetchPublicKeyResult;
+      }
+
+      if (retries < maxRetries) {
+        await new Promise((resolve) => setTimeout(resolve, delayInMs));
+      }
+    }
+
+    // After all retries, check if fetchPublicKeyResult was assigned and handle the case where it was not assigned (should not happen)
+    if (fetchPublicKeyResult == null) {
+      return errorResult({
+        errorMessage:
+          "Unexpected error in retry policy when fetching STS public keys",
+        errorCategory: "SERVER_ERROR",
+      });
+    }
+
+    return fetchPublicKeyResult;
+  }
+
   private readonly fetchPublicKey = async (
     stsJwksEndpoint: string,
   ): Promise<Result<JSON>> => {
```

Comment:
I think we can't say the type is JSON because we don't know that it actually is as `.json` returns `any` 🤔

Reply:
Feels counter intuitive but would it therefore be better to do response.text() and then parse it so we know it's JSON?

Reply:
Or do we make a public key type that we validate it's the shape of?

Reply:
After discussion will validate the keys